app/controllers/users/contact_details.js

```diff
@@ -0,0 +1,165 @@
+      }
+    },
+    actions: {
+      focusIn(field, value) {
+        this.set(`${field}previousValue`, value);
+      },
+
+      focusOut(field, value, required) {
+        if (value == this.get(`${field}previousValue`)) {
+          this.set(`${field}InputError`, false);
+          this.set("emailValidationError", false);
+          this.set("mobileValidationError", false);
+          return;
+        }
+
+        if (field === "mobile" && value.trim()) {
+          if (/^[456789]\d{7}/.test(value)) {
+            value = "+852" + value;
+          } else {
+            this.set("mobileValidationError", true);
+            this.set(
+              "user.mobileWithoutCountryCode",
+              this.get(`${field}previousValue`)
+            );
+            Ember.$("#mobile").focus();
+            return;
+          }
+        }
+
+        if (
+          field === "email" &&
+          value.trim() &&
+          !/^[^@\s]+@[^@\s]+/.test(value)
+        ) {
+          this.set("emailValidationError", true);
+          this.set(`user.${field}`, this.get(`${field}previousValue`));
+          Ember.$("#email").focus();
+          return;
+        }
+
+        if (!value.trim() && required) {
+          let displayValue =
+            field === "mobile" ? "mobileWithoutCountryCode" : field;
+          this.set(`${field}InputError`, true);
+          this.set(`user.${displayValue}`, this.get(`${field}previousValue`));
+          Ember.$(`#${field}`).focus();
+          return;
+        }
+        return this.runTask(async () => {
+          let newUser = this.get("store").peekRecord(
+            "user",
+            this.get("user.id")
+          );
+          newUser.set(field, value);
+          await newUser.save();
+          this.set(`${field}InputError`, false);
+          this.set("emailValidationError", false);
+          this.set("mobileValidationError", false);
+        }, ERROR_STRATEGIES.MODAL);
+      },
+
+      uploadEditedImageSuccess(e, data) {
+        var identifier =
+          data.result.version +
+          "/" +
+          data.result.public_id +
+          "." +
+          data.result.format;
+        var newUploadedImage = this.get("store").createRecord("image", {
+          cloudinaryId: identifier,
+          favourite: true
+        });
+        this.set("newUploadedImage", newUploadedImage);
+        this.set("userImageKeys", identifier);
+        this.send("saveEditedImage");
+      },
+
+      async saveEditedImage() {
+        return this.runTask(async () => {
+          let newUser = this.get("store").peekRecord(
+            "user",
+            this.get("user.id")
+          );
+          newUser.set("image", await this.saveImage());
+          newUser.save();
+        }, ERROR_STRATEGIES.MODAL);
+      },
+
+      deleteImage() {
+        this.runTask(async () => {
+          let newUser = this.get("store").peekRecord(
+            "user",
+            this.get("user.id")
+          );
```

Comment:
we can use `user` value defines above (line 12)
https://github.com/crossroads/stock.goodcity/pull/855/files#diff-c94f26135328a1f7cc107818ba4d2aa7R12

Reply:
```suggestion
          let user = this.get("user");
```